dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -399,6 +408,58 @@ private static bool IsAsyncMethod(MethodInfo method)
         return (parameterFunc, parameterView);
     }
 
+    /// <summary>
+    /// Tries to deserialize the given value into an object of the specified target type.
+    /// </summary>
+    /// <param name="value">The value to be deserialized.</param>
+    /// <param name="targetType">The type of the object to deserialize the value into.</param>
+    /// <param name="deserializedValue">The deserialized object if the method succeeds; otherwise, null.</param>
+    /// <returns>true if the value is successfully deserialized; otherwise, false.</returns>
+    private static bool TryToDeserializeValue(object value, Type targetType, out object? deserializedValue)
+    {
+        try
+        {
+            // Handling known 'JSON' types.
+            if (value is JsonDocument document)
+            {
+                deserializedValue = JsonSerializer.Deserialize(document, targetType);
+                return true;
+            }
+
+            if (value is JsonNode node)
+            {
+                deserializedValue = JsonSerializer.Deserialize(node, targetType);
+                return true;
+            }
+
+            if (value is JsonElement element)
+            {
+                deserializedValue = JsonSerializer.Deserialize(element, targetType);
+                return true;
+            }
+
+            // The JSON can be represented by other data types from various libraries. For example, JObject, JToken, and JValue from the Newtonsoft.Json library.  
+            // Since we don't take dependencies on these libraries and don't have access to the types here,
+            // the only way to deserialize those types is to convert them to a string first by calling the 'ToString' method.
+            // Attempting to use the 'JsonSerializer.Serialize' method, instead of calling the 'ToString' directly on those types, can lead to unpredictable outcomes.
+            // For instance, the JObject for { "id": 28 } JSON is serialized into the string  "{ "Id": [] }", and the deserialization fails with the
+            // following exception - "The JSON value could not be converted to System.Int32. Path: $.Id | LineNumber: 0 | BytePositionInLine: 7."
+            deserializedValue = JsonSerializer.Deserialize(value.ToString(), targetType);
+            return true;
```

Comment:
nit: a little simplification for try block, feel free to skip:
```csharp
deserializedValue = value switch
{
    JsonDocument document => document.Deserialize(targetType),
    JsonNode node => node.Deserialize(targetType),
    JsonElement element => element.Deserialize(targetType),
    _ => JsonSerializer.Deserialize(value.ToString(), targetType)
};

return true;
```

Reply:
Yep, thanks.